setup/populate_surveys.rb

```diff
@@ -1,19 +1,77 @@
 # frozen_string_literal: true
 ActiveRecord::Base.logger = Logger.new(STDOUT)
 
+def populate_courses
+  clear_courses
+  populate_users
+  
+  n_courses = ENV['courses'] || 100
+  to_insert = []
+  n_courses.to_i.times do |i|
+    to_insert << {
+      title: "generated course",
+      slug: "course-#{i}",
+    }
+  end
+  Course.insert_all(to_insert)
+  courses_users = []
+  to_insert.each_with_index do |course, i|
+    course_id = Course.find_by(slug: "course-#{i}").id
+    1..20.times do |i|
```

Comment:
A more efficient and Rails-y way to do this would be to fetch the random set of users in one query and then iterate through them. Since you just created the courses and this is a dev environment where we can assume that no other database transactions are happening at the same time, you could just do `Course.last(n_courses)` and iterate through them.then you don't need to worry about an iterator either.

Also, you have a nested loop here, but both the inner and outer loop use the same index variable. In terms of scoping this is not a problem, but it's good practice to always use separate names (eg, i, j, m, n) in cases like this for readability.